build issue on Linux ("incompatible pointer type")

Attempting to build from the latest source on Arch Linux gives this error message for both the d1 and d2 codebases:

`[ 80%] Building C object main/CMakeFiles/d1x-redux.dir/multibot.c.o
/home/hem/sources/dxx-redux/d1/main/menu.c: In function ‘menu_obs_options_handler’:
/home/hem/sources/dxx-redux/d1/main/menu.c:2370:101: error: passing argument 3 of ‘newmenu_listbox1’ from incompatible pointer type [-Wincompatible-pointer-types]
 2370 |                                 newmenu_listbox1("Select Game Mode", SDL_arraysize(Obs_mode_names), Obs_mode_names,
      |                                                                                                     ^~~~~~~~~~~~~~
      |                                                                                                     |
      |                                                                                                     const char **
In file included from /home/hem/sources/dxx-redux/d1/main/menu.c:48:
/home/hem/sources/dxx-redux/d1/main/newmenu.h:148:65: note: expected ‘char **’ but argument is of type ‘const char **’
  148 | extern listbox *newmenu_listbox1(char *title, int nitems, char *items[], int allow_abort_flag, int default_item, int (*listbox_callback)(listbox *lb, d_event *event, void *userdata), void *userdata);
      |                                                           ~~~~~~^~~~~~~
make[2]: *** [main/CMakeFiles/d1x-redux.dir/build.make:566: main/CMakeFiles/d1x-redux.dir/menu.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:410: main/CMakeFiles/d1x-redux.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
`

--- a/d1/main/menu.c
+++ b/d1/main/menu.c
@@ -2367,7 +2367,7 @@ int menu_obs_options_handler ( newmenu *menu, d_event *event, void *userdata )
 
 		case EVENT_NEWMENU_SELECTED:
 			if (citem == 1) {
-				newmenu_listbox1("Select Game Mode", SDL_arraysize(Obs_mode_names), Obs_mode_names,
+				newmenu_listbox1("Select Game Mode", SDL_arraysize(Obs_mode_names), (char **)Obs_mode_names,
 					1, cmode, select_obs_game_mode_handler, obs_menu_data);
 				return 1;
 			}

--- a/d2/libmve/mveplay.c
+++ b/d2/libmve/mveplay.c
@@ -485,7 +485,7 @@ static int audio_data_handler(unsigned char major, unsigned char minor, unsigned
 				// copy back to the audio buffer
 				mve_free(mve_audio_buffers[mve_audio_buftail]); // free the old audio buffer
 				mve_audio_buflens[mve_audio_buftail] = cvt.len_cvt;
-				mve_audio_buffers[mve_audio_buftail] = cvt.buf;
+				mve_audio_buffers[mve_audio_buftail] = (short *)cvt.buf;
 			}
 #endif
 

--- a/d2/main/menu.c
+++ b/d2/main/menu.c
@@ -2404,7 +2404,7 @@ int menu_obs_options_handler ( newmenu *menu, d_event *event, void *userdata )
 
 		case EVENT_NEWMENU_SELECTED:
 			if (citem == 1) {
-				newmenu_listbox1("Select Game Mode", SDL_arraysize(Obs_mode_names), Obs_mode_names,
+				newmenu_listbox1("Select Game Mode", SDL_arraysize(Obs_mode_names), (char **)Obs_mode_names,
 					1, cmode, select_obs_game_mode_handler, obs_menu_data);
 				return 1;
 			}

